spacepy/time.py

```diff
@@ -1409,6 +1438,52 @@ def getGPS(self):
             self.TAI - GPS0, attrs={'dtype': 'GPS'})
         return self.GPS
 
+    # -----------------------------------------------
+    def getAPT(self):
+        """
+        a.APT or a.getAPT()
+
+        Return AstroPy time object.
+
+        Always recalculates from the current value of ``TAI``, which
+        will be created if necessary.
+
+        Updates the ``APT`` attribute.
+
+        Returns
+        ========
+            out : astropy.time.Time
+                AstroPy Time object
+
+        Notes
+        =====
+        .. versionadded:: 0.2.2
+
+        Requires AstroPy 1.0.
+
+        Examples
+        ========
+        >>> a = Ticktock('2002-02-02T12:00:00', 'ISO')
+        >>> a.APT
+        <Time object: scale='utc' format='gps' value=696686413.0>
+
+        See Also
+        =========
+        getUTC, getUNX, getRDT, getJD, getMJD, getCDF, getISO, getDOY, geteDOY,
+        getGPS
+        """
+        if self.data.attrs['dtype'] == 'APT':
+            # This should be the case from the constructor
+            self.APT = self.data
+            return self.APT
+        if not HAVE_ASTROPY:
+            raise RuntimeError('Import of astropy.time failed.')
+        GPS0 = 694656019
+        self.APT = astropy.time.Time(
+            self.TAI - GPS0, scale='tai', format='gps')
```

Comment:
Do we want to output on the `tai` scale? Or the `utc` scale?
I was somewhat confused when I tried this out initially and printed an ISO time that was significantly different to the input I used. That, of course, was fixed when I changed the _scale_ to UTC. I believe that GPS time is the same on each scale (where the ISO time, JD, etc.) are not. For a user, it might be more intuitive to output an `astropy.time.Time` object on the `utc` scale...
Related question, do we still want to do this as `self.TAI - GPS0` or just `self.GPS`? They should be the same, right?

Reply:
I agree that seeing an ISO datestamp for a TAI time is confusing to space physicists :)

The constructor for Time takes the input scale, not the output scale. The AstroPy docs claim that GPS time is intrinsically on the TAI scale. I've checked and it's possible to specify the input scale as `utc` and format as `gps`, and it seems to give equivalent answers:
```python
t = astropy.time.Time(5e8, format='gps', scale='tai')
t = t.utc
t.format='iso'
print(t) #1995-11-10 00:53:10.000
t2 = astropy.time.Time(5e8, format='gps', scale='utc')
t2.format='iso'
print(t2) #1995-11-10 00:53:10.000
t3 = spacepy.time.Ticktock(5e8, dtype='GPS')
print(t3.iSO[0]) #1995-11-10T00:53:10
```
However lacking a really good definition of what AstroPy considers to be the `utc` "scale" makes me nervous. E.g. they claim the unix format as being on the `utc` scale, which would involve pretending leapseconds are moments when time doesn't exist.

We could be sure by, for instance, doing:
```python
self.APT = astropy.time.Time(self.TAI - GPS0, scale='tai', format='gps').utc
```
but making two Time objects seems nasty, and I'm nervous about potentially losing leapseconds there. (It probably _won't_, mind you, just nervous...) Doing `gps` format, `tai` scale seemed the best way to make sure the information in the Time object 100% corresponded to the Ticktock input, and presumably a user working with AstroPy Time can perform manipulations to their desired output format (and scale).

`self.TAI - GPS0` should be the same as `self.GPS`, but by doing the conversion explicitly from TAI, I'm avoiding some possibility of loops doing the lookup. The interactions between getXXX, getattr, init, and update_items are tricky. There's special sauce that prioritizes doing TAI first and (to a lesser extent) UTC second, and everything else coming from one of those. Elevating GPS to that level seemed to invite complexity to save copy/pasting a subtraction.